prow/plank/controller.go

```diff
@@ -167,41 +160,42 @@ func (c *Controller) Sync() error {
 	close(pjCh)
 
 	errCh := make(chan error, len(pjs))
+	reportCh := make(chan kube.ProwJob, len(pjs))
 	wg := &sync.WaitGroup{}
 	wg.Add(maxSyncRoutines)
 	for i := 0; i < maxSyncRoutines; i++ {
-		go c.syncProwJob(wg, pjCh, errCh, pm)
+		go c.syncProwJob(wg, pjCh, errCh, reportCh, pm)
 	}
 	wg.Wait()
 	close(errCh)
+	close(reportCh)
 
 	for err := range errCh {
 		syncErrs = append(syncErrs, err)
 	}
 
 	var reportErrs []error
-	c.reportsLock.Lock()
-	for _, pj := range c.reports {
-		if err := c.report(pj); err != nil {
+	for report := range reportCh {
```

Comment:
We can also do this in parallel, wanted to keep a minimal diff unless we had a reason to though